### DISCUSSION TITLE
Using Apache AGE for RDF/OWL Ontology TTL Export from Protege

### DISCUSSION THREAD

**Discussion by modlibrarian** [apache/age#1788] [Q&A]
Hello. I am interested in using a taxonomy/ontology tool like Protege or Graphologi to create the nodes and relationships for our topic graph. Our engineering team wants to use Apache AGE as the graph visualizer. Does Apache AGE support import of a TTL file format?

**Comment by markgomer** [comment]
Thank you for your interest, @modlibrarian. 
As far as I know, Apache AGE only supports importing .csv files: 
https://age.apache.org/age-manual/master/intro/agload.html

**Comment by dpdjvhxm** [comment]
@modlibrarian I'm not sure about TTL file format well.
But I think you can translate TTL to .csv file, then try it

This step will be help to translate :
1. Data Extraction: Extract RDF triples (subject, predicate, object) from TTL files.
2. Structure: Structure the extracted data into rows and columns. At this point, you need to decide how to assign each triple to a row in the CSV.
3. Save in CSV format: Save structured data as a CSV file.

**Reply by markgomer** [reply]
There seems to exist many tools to convert between those formats, too. 

This is the module on AGE source code to load .csv files: https://github.com/apache/age/tree/master/src/backend/utils/load
one could use it as a model to add a .ttl import function to it.
